examples/nlp/text_classification/conf/config.yaml

```diff
@@ -0,0 +1,81 @@
+# Sentence classification with pretrained BERT models
+
+io:
+  work_dir: ./outputs
+  checkpoint_dir: null
+  save_epoch_freq: 1
+  save_step_freq: -1
+
+pl:
+  trainer:
+    gpus: 1 # the number of gpus, 0 for CPU
+    num_nodes: 1
+    max_epochs: 10
+    max_steps: null # precedence over max_epochs
+    accumulate_grad_batches: 1 # accumulates grads every k batches
+    amp_level: O0 # O1/O2 for mixed precision
+    distributed_backend: ddp
+
+model:
+  data_dir: ??? # /path/to/data
+  class_balancing: null # or weighted_loss
+
+  train_ds:
+    prefix: train
+    batch_size: 32
+    shuffle: true
+    num_samples: -1
+    num_workers: 2
+    pin_memory: false
+    use_cache: true
```

Comment:
When use_cache is true, if data or processing params are changed, does it detect such changes and let the user know?

Reply:
We can add any sort of logging to ```text_classification_datasets.py``` but I don't think we need to do this in this PR.

Reply:
If I understand how it works, it may introduce bugs if user is not aware of it. Currently our datasets are small and it is very fast to create them. For instance, user should know that when she changes the max_length, or num of samples, etc (anything related to dataset), it not effective as they are cached. Did I get correctly how caching works?